pkg/controller/linstorsatelliteset/linstorsatelliteset_controller.go

```diff
@@ -158,7 +161,8 @@ func (r *ReconcileLinstorSatelliteSet) Reconcile(request reconcile.Request) (rec
 
 	r.linstorClient, err = lc.NewHighLevelLinstorClientFromConfig(pns.Spec.ControllerEndpoint, &pns.Spec.LinstorClientConfig, getSecret)
 	if err != nil {
-		return reconcile.Result{}, err
```

Comment:
This will cause issues when there is an error because `r.linstorClient` will be `nil` but it is used later on.

Reply:
Good catch! I'll just update the status right there.